Source/WebCore/platform/mediarecorder/cocoa/AudioSampleBufferCompressor.mm

```diff
@@ -94,23 +95,41 @@
     return true;
 }
 
+Ref<GenericPromise> AudioSampleBufferCompressor::drain()
```

Comment:
Will we have a use for flushInternal(false), or will we use drain() instead?

Reply:
no we still have a use for `flush()` (`flushInternal(false)`)

a "drain" pulls all samples from the AudioConverter, including its internal buffer, it it no longer usable after that unless reset.

So basically if we are about to record an ongoing audio stream, we'll use flush() and wait if needed for a future audio sample which would output the current one.

A call to drain() is when we're paused, and no more audio is coming. When re-submitting samples to the encoder after a drain, nothing will come out of it for about 50ms (for AAC it generates three frames with the same timestamp that are marked as having skip for 50ms)